### ISSUE TITLE
Commit 6dca5715672fa66ccc1e6d0fbc6f3ff28370896d causes segfault

### ISSUE THREAD

**Issue by lemenkov** [hufrea/byedpi#213]
I've got this  error with recent byedpi:

```
recv: Bad file descriptor
...
recv: Bad file descriptor
recv: Bad file descriptor
recv: Bad file descriptor
recv: Bad file descriptor
recv: Bad file descriptor
recv: Bad file descriptor
recv: Bad file descriptor
ciadpi: conev.c:49: add_event: Assertion `fd > 0' failed.
run.sh: строка 3: 17637 Аварийный останов         (образ памяти сброшен на диск) ./ciadpi -i 127.0.0.1 -p 10800 --oob 1
```

Afer git bisection I've found an offender - 6dca5715672fa66ccc1e6d0fbc6f3ff28370896d.

**Comment by hufrea** [comment]
How to reproduce this?

**Comment by lemenkov** [comment]
(reverting 6dca5715672fa66ccc1e6d0fbc6f3ff28370896d fixed this btw)

**Comment by hufrea** [comment]
Logs, please (`--debug 2`)

**Comment by zimbabwe303** [comment]
I also cannot reproduce this. Tried different sites and it still works okay.

**Comment by lemenkov** [comment]
Folks, I can't say what's changed but I can't reproduce it since yesterday. I'll try to play with it for a while (got a few ideas to try).

**Comment by denizzzka** [comment]
Yesterday I run into same issue after ~2 weeks of normal run. And I just restarted service instead of run debugger
Damn!

**Comment by denizzzka** [comment]
Looks like same? But on current master: https://github.com/hufrea/byedpi/issues/228#issuecomment-2466831471

**Comment by lemenkov** [comment]
I believe it was addressed in PR  #229.
